### ISSUE TITLE
media-platform-js-sdk configures retries for global axios instance

### ISSUE THREAD

**Issue by alexeevg** [wix/media-platform-js-sdk#90]
https://github.com/wix/media-platform-js-sdk/blob/b3af728c0b92d595df639f78f2a80a74126b8405/src/platform/http/http-client.ts#L63

This is a problem for a number of reasons. The app using media-platform may use global axios and not expect that all requests (even non-idempotent ones, e.g. POST requests) will be retried. Moreover, `axios-retry` has a very broad idea of retryable error. Consider this case (which happened to us):

```
await axios.post(url, createReadStream('this-file-does-not-exist.jpg'));
```

Instead of failing fast it hangs forever, because on the first attempt the file stream emits an error, `axios-retry` invokes a second attempt with the same file stream in a broken state.

The bottom line is: IMHO `media-platform-js-sdk` should not mess with global settings and configure a private axios instance instead.
